modules/improvedigitalBidAdapter.js

```diff
@@ -0,0 +1,414 @@
+
+        for (var bidNumber = 0; bidNumber < bidObjects.length; bidNumber++) {
+          let bidObject = bidObjects[bidNumber];
+
+          if (bidObject.id === bidRequest.bidId) {
+            let bid;
+            if (!bidObject.price || bidObject.price === null) {
+              bid = bidfactory.createBid(CONSTANTS.STATUS.NO_BID);
+              bid.bidderCode = IMPROVE_DIGITAL_BIDDER_CODE;
+              bidmanager.addBidResponse(bidRequest.placementCode, bid);
+              continue;
+            }
+            if (bidObject.errorCode && bidObject.errorCode !== 0) {
+              bid = bidfactory.createBid(CONSTANTS.STATUS.NO_BID);
+              bid.bidderCode = IMPROVE_DIGITAL_BIDDER_CODE;
+              bidmanager.addBidResponse(bidRequest.placementCode, bid);
+              continue;
+            }
+            if (!bidObject.adm || bidObject.adm === null || typeof bidObject.adm !== 'string') {
+              bid = bidfactory.createBid(CONSTANTS.STATUS.NO_BID);
+              bid.bidderCode = IMPROVE_DIGITAL_BIDDER_CODE;
+              bidmanager.addBidResponse(bidRequest.placementCode, bid);
+              continue;
+            }
+
+            bid = bidfactory.createBid(CONSTANTS.STATUS.GOOD);
+
+            let syncString = '';
+            let syncArray = (bidObject.sync && bidObject.sync.length > 0) ? bidObject.sync : [];
+
+            for (var syncCounter = 0; syncCounter < syncArray.length; syncCounter++) {
+              syncString += (syncString === '') ? 'document.writeln(\"' : '';
+              let syncInd = syncArray[syncCounter];
+              syncInd = syncInd.replace(/\//g, '\\\/');
+              syncString += '<img src=\\\"' + syncInd + '\\\"\/>';
+            }
+            syncString += (syncString === '') ? '' : '\")';
+
+            let nurl = '';
+            if (bidObject.nurl && bidObject.nurl.length > 0) {
+              nurl = '<img src=\"' + bidObject.nurl + '\" width=\"0\" height=\"0\" style=\"display:none\">';
+            }
+            bid.ad = nurl + '<script>' + bidObject.adm + syncString + '</script>';
+            bid.bidderCode = IMPROVE_DIGITAL_BIDDER_CODE;
+            bid.cpm = parseFloat(bidObject.price);
+            bid.width = bidObject.w;
+            bid.height = bidObject.h;
+
+            bidmanager.addBidResponse(bidRequest.placementCode, bid);
+          }
+        }
+      }
+    }
+  };
+
+  baseAdapter.callBids = function (params) {
+    // params will contain an array
+    let bidRequests = params.bids || [];
+    let loc = utils.getTopWindowLocation();
+    let requestParameters = {
+      singleRequestMode: false,
+      httpRequestType: this.idClient.CONSTANTS.HTTP_REQUEST_TYPE.GET,
+      callback: CALLBACK_FUNCTION,
+      secure: (loc.protocol === 'https:') ? 1 : 0,
+      libVersion: this.LIB_VERSION
+    };
+
+    let normalizedBids = [];
+    for (var bidCounter = 0; bidCounter < bidRequests.length; bidCounter++) {
```

Comment:
You can use a https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map over here and all the other places in this file. No need to create an array and push..  your code will be much cleaner.

```
let normalizedBids = bidRequest.map((bidRequest) => {
    // your code
    return result;
});

```